### ISSUE TITLE
Focus styles: Select color for focus style and identify other focus methods, if necessary

### ISSUE THREAD

**Issue by davidakennedy** [department-of-veterans-affairs/vets-design-system-documentation#1314]
## Context

The VA.gov Design System needs updated focus styles for better accessibility because the current styles don't meet color contrast guidelines in the Web Content Accessibility Guidelines. In updating the styles, we want to also think ahead to future versions of WCAG, which likely have more stringent requirements for focus styles.

Specifically, the current focus style fails to meet compliance in the scenario of white button on dark background (e.g. the active search button).

Links:

- [Success Criterion 1.4.11 Non-text Contrast](https://www.w3.org/TR/WCAG21/#non-text-contrast)
- [What's New in WCAG 2.2?](https://www.w3.org/WAI/standards-guidelines/wcag/new-in-22/)
- [2.4.11 Focus Appearance(AA)](https://www.w3.org/WAI/standards-guidelines/wcag/new-in-22/#2411-focus-appearance-aa)
- [2.4.12 Focus Not Obscured (Minimum) (AA)](https://www.w3.org/WAI/standards-guidelines/wcag/new-in-22/#2412-focus-not-obscured-minimum-aa)
- [Testing, Evaluation, and Documentation: Double box-shadow focus styling
](https://docs.google.com/document/d/1o-bfmdz6arPju9XCiGn-NR9UpgoN1Vp0EB3cvRGMA3Q/edit?usp=sharing)
- [VA Design System Focus Styles: Questions and Assumptions](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/shared-support/accessibility/projects/design-system-focus-styles/focus-styles-questions-assumptions-requirements.md)
- [Use double box-shadow for focus styling #654](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/654)

## Work to be done

This work is to:

- select a color for the double box shadow focus method. - [Colors should be from this palette](https://designsystem.digital.gov/design-tokens/color/system-tokens/).
- explore alternative focus methods, if needed for other elements, and select a color for it (will be the same as the above).
- create demos for new focus styles, so they can be tested and evaluated by stakeholders for feedback. 

Feedback needed:

- Does the color work well in the stress cases identified in  [Testing, Evaluation, and Documentation: Double box-shadow focus styling](https://docs.google.com/document/d/1o-bfmdz6arPju9XCiGn-NR9UpgoN1Vp0EB3cvRGMA3Q/edit?usp=sharing)?
- Does the color harmonize well with other visual elements on VA.gov?

## Acceptance criteria

- [ ] Color recommendations selected, at least two options.
- [ ] Alternative focus method explored, at least two alternatives explored.
- [ ] Demos created showing both methods with colors.
- [ ] All artifacts shared with stakeholders and/or added to tickets.

**Comment by davidakennedy** [comment]
This work isn't complete, but I've started thinking through some possible color options and wanted to share and explain the thinking here. The color explorations can be seen in the [audit recommendations and notes doc](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/shared-support/accessibility/projects/design-system-focus-styles/focus-style-audit-recommendations-notes.md) I created. Some thoughts on why I went the direction I did in starting with those colors:

- I steered clear of blue colors because so much of VA.gov is already blue. So a blue focus style wouldn't stand out very well in a sea of blue.
- All of the colors fall into the 50 range on the [color token color scale](https://designsystem.digital.gov/design-tokens/color/system-tokens/). My reasoning there is that the colors in this range have a 4.6:1 contrast ratio on white. That means we'd have some wiggle room for the contrast to still meet the required 3:1 needed for interface elements on background colors like light blue or light yellow, found in Alerts and elsewhere. The next step up in colors (the 40 range) all have a contrast of 3.2:1 on white. That means once they get on any other color, they'd likely wouldn't meet contrast ratio requirements.

A few other thoughts on where I'm at with this work:

- These colors are only starting points. They need to be explored further and put through testing on elements.
- I'd recommend testing them with a color blindness simulator, creating a color palette with a [tool like this](https://line47.com/accessible-color-matrix/) or similar, and testing them in the browser on real content on the VA's site.
- I'd also recommending creating a PR with Storybook to help visualize how the colors and implementation techniques will look on components in the design system.

**Comment by humancompanion-usds** [comment]
@davidakennedy - While working on the Back-to-Top change from button to link I came across the Scottish Design System's focus state (Laura, I think, also pointed me to it in another issue; great minds and all that). Instead of using a border ring for focus on buttons and links it flips the colors and adds a bottom inset drop shadow as a border. Worth a look I think: https://designsystem.gov.scot/styles/states/

**Comment by davidakennedy** [comment]
@danbrady Thanks for the report! I'm not actively working on this, but definitely encourage you all to drop gotchas in this ticket so the team has it when work begins again.

**Comment by davidakennedy** [comment]
I've started to work on this between other team work. A few notes:

- All of this is in progress, and I haven't landed on any recommendations yet. Sharing here for transparency.
- I've gathered some initial feedback from accessibility specialists, but hope to share progress soon with the design system team.
- This [repo has my ongoing notes and code](https://github.com/department-of-veterans-affairs/vads-focus-styles). The demo page is linked from there, but you [can access it here too](https://silver-adventure-6k44jeq.pages.github.io/).
- I'm going to continue building out demos with the other colors and share it with the team soon.

**Comment by caw310** [comment]
DK presented this to DSC during the Dec 19 meeting. 
This ticket is in the DST backlog. 

DSC meeting notes: https://docs.google.com/document/d/15Sg_cifRNJMhseuUC6dxktQ0cLU5HidZ7OFeB2Djxpw/edit?usp=sharing

**Comment by davidakennedy** [comment]
Thanks, Carol!

Some takeaways for me on next steps:

- Go through the design system PR and look for any focus styles on components that could use more work/need tweaking. Note those.
- Iterate on the checkbox/radio button styles. Particularly, the radio buttons look smaller when focused. Maybe move the black outline to the outside, rather than inside.
- Start compiling implementation notes for the design system team.

**Comment by davidakennedy** [comment]
Some updates here from the last few weeks:

- I completed [the implementation notes](https://github.com/department-of-veterans-affairs/vads-focus-styles/blob/gh-pages/implementation-notes.md), which also have the notes to help iterate on the in-progress PR.
- I completed the [additional explorations on checkboxes and radio buttons](https://github.com/department-of-veterans-affairs/vads-focus-styles/tree/gh-pages?tab=readme-ov-file#additional-explorations).
- I had a [meeting with the design system team](https://github.com/department-of-veterans-affairs/vads-focus-styles/tree/gh-pages?tab=readme-ov-file#notes-from-2025-01-16-design-system-team-standup) to help close things out.

This should be good to close out now, and the design system team can pick it up when ready. cc: @caw310
